x/confidentialtransfers/keeper/msg_server_test.go

```diff
@@ -456,6 +478,38 @@ func (suite *KeeperTestSuite) TestMsgServer_DepositTooManyPendingBalances() {
 	suite.Require().ErrorContains(err, "account has too many pending transactions")
 }
 
+// Tests that the Deposit method fails when the feature is disabled.
+func (suite *KeeperTestSuite) TestMsgServer_DepositFeatureDisabled() {
```

Comment:
I think we may not need half of init steps here, that distract from actual test. This works and might apply to other tests:
```
func (suite *KeeperTestSuite) TestMsgServer_DepositFailsIfFeatureDisabled() {
	// Disable the confidential tokens module via params
	suite.Ctx = suite.App.BaseApp.NewContext(false, tmproto.Header{})
	params := types.DefaultParams()
	params.EnableFeature = false
	suite.App.ConfidentialTransfersKeeper.SetParams(suite.Ctx, params)

	_, err := suite.msgServer.Deposit(sdk.WrapSDKContext(suite.Ctx), &types.MsgDeposit{})
	suite.Require().Error(err, "Should not be able to deposit when feature is disabled")
	suite.Require().ErrorContains(err, "feature is disabled by governance")
}
```

Reply:
Yup you're right that based on the way the code is written now there's technically no need to initialize (since the IsModuleEnabled is the first check) but wanted to write the tests in a way that the only reason why it would fail is because the flag was not doing it's job (and not due to something else)

The last line 
`suite.Require().ErrorContains(err, "feature is disabled by governance")` does check the failure reason is correct, but I thought it seems a bit flaky to rely on the order of checks:

For example, if in the future we added a check to see if the account exists first, before checking the CtModuleEnabled flag, then this test would fail (even though the flag is turned off and would have been working as expected).